pypeit/core/flux_calib.py

```diff
@@ -66,17 +66,13 @@ def blackbody_func(a, teff):
         a (float):
             flux normalisation factor
         teff (float):
-            Effective temperature of the blackbody
+            Effective temperature of the blackbody (in units of K)
```

Comment:
If I understand what you've done in this function teff is not a float, but rather an astropy Quantity. Please avoid making satrapy quantities inputs to functions like this. I would make the inputs floats and then clearly specify what their units are in the docs. We have a convention in PypeIt which is that we don't use astropy Quantities except internally in functions and then we also pass back floats (i.e. .value) and not quantities. This is done for very good reasons, namely computations with astropy Quantities can slow down floating point operations by a factor of 10 or more. My suggestion is thus that you pass in floats and pass out floats, and only use the quantities and units internally. I love this functionality in astropy, but I think this is the best approach given how slow it is.